src/capella/beacon_state.rs

```diff
@@ -0,0 +1,68 @@
+};
+use crate::primitives::{
+    Bytes32, Gwei, ParticipationFlags, Root, Slot, ValidatorIndex, WithdrawalIndex,
+};
+use ssz_rs::prelude::*;
+
+#[derive(Default, Debug, SimpleSerialize, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct BeaconState<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const SYNC_COMMITTEE_SIZE: usize,
+    const BYTES_PER_LOGS_BLOOM: usize,
+    const MAX_EXTRA_DATA_BYTES: usize,
+    const MAX_BYTES_PER_TRANSACTION: usize,
+    const MAX_TRANSACTIONS_PER_PAYLOAD: usize,
+> {
+    #[serde(with = "crate::serde::as_string")]
+    pub genesis_time: u64,
+    pub genesis_validators_root: Root,
+    #[serde(with = "crate::serde::as_string")]
+    pub slot: Slot,
+    pub fork: Fork,
+    pub latest_block_header: BeaconBlockHeader,
+    pub block_roots: Vector<Root, SLOTS_PER_HISTORICAL_ROOT>,
+    pub state_roots: Vector<Root, SLOTS_PER_HISTORICAL_ROOT>,
+    pub historical_roots: List<Root, HISTORICAL_ROOTS_LIMIT>,
+    pub eth1_data: Eth1Data,
+    pub eth1_data_votes: List<Eth1Data, ETH1_DATA_VOTES_BOUND>,
+    #[serde(with = "crate::serde::as_string")]
+    pub eth1_deposit_index: u64,
+    pub validators: List<Validator, VALIDATOR_REGISTRY_LIMIT>,
+    #[serde(with = "crate::serde::collection_over_string")]
+    pub balances: List<Gwei, VALIDATOR_REGISTRY_LIMIT>,
+    pub randao_mixes: Vector<Bytes32, EPOCHS_PER_HISTORICAL_VECTOR>,
+    #[serde(with = "crate::serde::collection_over_string")]
+    pub slashings: Vector<Gwei, EPOCHS_PER_SLASHINGS_VECTOR>,
+    #[serde(with = "crate::serde::collection_over_string")]
+    pub previous_epoch_participation: List<ParticipationFlags, VALIDATOR_REGISTRY_LIMIT>,
+    #[serde(with = "crate::serde::collection_over_string")]
+    pub current_epoch_participation: List<ParticipationFlags, VALIDATOR_REGISTRY_LIMIT>,
+    pub justification_bits: Bitvector<JUSTIFICATION_BITS_LENGTH>,
+    pub previous_justified_checkpoint: Checkpoint,
+    pub current_justified_checkpoint: Checkpoint,
+    pub finalized_checkpoint: Checkpoint,
+    #[serde(with = "crate::serde::collection_over_string")]
+    pub inactivity_scores: List<u64, VALIDATOR_REGISTRY_LIMIT>,
+    pub current_sync_committee: SyncCommittee<SYNC_COMMITTEE_SIZE>,
+    pub next_sync_committee: SyncCommittee<SYNC_COMMITTEE_SIZE>,
+    pub latest_execution_payload_header: ExecutionPayloadHeader<
+        BYTES_PER_LOGS_BLOOM,
+        MAX_EXTRA_DATA_BYTES,
+        MAX_BYTES_PER_TRANSACTION,
+        MAX_TRANSACTIONS_PER_PAYLOAD,
+    >,
+    pub next_withdrawal_index: WithdrawalIndex,
+    pub next_withdrawal_validator_index: ValidatorIndex,
```

Comment:
```suggestion
     #[serde(with = "crate::serde::as_string")]
    pub next_withdrawal_index: WithdrawalIndex,
     #[serde(with = "crate::serde::as_string")]
    pub next_withdrawal_validator_index: ValidatorIndex,
```

Reply:
@ralexstokes thanks for the suggestions and sorry to have missed these! I should've really tested the serialisation / deserialisation output myself - I'll do another pass to make sure I haven't miss anything.